src/redux/books/books.js

```diff
@@ -1,25 +1,48 @@
+import { createAsyncThunk } from '@reduxjs/toolkit';
+import axios from 'axios';
+
 const ADD_BOOK = 'books/books/ADD_BOOK';
 const REMOVE_BOOK = 'books/books/REMOVE_BOOK';
+const SHOW_BOOKS = 'books/books/SHOW_BOOKS';
+const baseURL = 'https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/C6d01TDYTG44NnpUlkMZ/books';
+
+export const fetchBooks = createAsyncThunk(
+  SHOW_BOOKS,
+  async (_args, { dispatch }) => {
+    const { data } = await axios.get(baseURL);
+    const books = Object.keys(data).map((key) => {
+      const book = data[key][0];
+      return {
+        item_id: key,
+        ...book,
+      };
+    });
+    dispatch({ type: SHOW_BOOKS, payload: books });
+  },
+);
 
-export const addBook = (data) => (dispatch) => {
-  dispatch({ type: ADD_BOOK, payload: data });
+export const addBook = (data) => async (dispatch) => {
+  await axios.post(baseURL, data);
+  dispatch({ type: ADD_BOOK });
+  dispatch(fetchBooks());
 };
```

Comment:
- [OPTIONAL] I am only able to add just one book object. I got an error message for another attempt after the first book object added. I would recommend you to look into it and fix. Take a look.

[app-behavior.webm](https://user-images.githubusercontent.com/26597432/208919140-6fdc9948-00ea-4675-81c5-da56362edb71.webm)

Reply:
@Qoosim Thanks for the review. At my end, I can post multiple data to the API. If you check the API URL you can see the data. https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/C6d01TDYTG44NnpUlkMZ/books